From cb472820e80a3e20c1f4468cd775f2b217220a4e Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Fri, 10 Nov 2023 22:02:12 +0100
Subject: [PATCH] CMakeLists.txt: Changes to fix TMPDIR injections

Some Random fixes to do not let the TMPDIR / builddir inject into the
artifacts.

Upstream-Status: inappropriate (oe specific)

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeLists.txt          |  1 -
 CMakeModules/ompl.pc.in |  2 +-
 omplConfig.cmake.in     | 18 +++++++-----------
 src/ompl/CMakeLists.txt |  3 ++-
 4 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09fc750..f52e4e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,6 @@ set_package_properties(Eigen3 PROPERTIES
     URL "http://eigen.tuxfamily.org"
     PURPOSE "A linear algebra library used throughout OMPL.")
 find_package(Eigen3 REQUIRED)
-include_directories("${EIGEN3_INCLUDE_DIR}")
 
 set_package_properties(MORSE PROPERTIES
     URL "https://www.openrobots.org/wiki/morse"
diff --git a/CMakeModules/ompl.pc.in b/CMakeModules/ompl.pc.in
index b7bbe3d..c6bffdc 100644
--- a/CMakeModules/ompl.pc.in
+++ b/CMakeModules/ompl.pc.in
@@ -9,4 +9,4 @@ Description: @PKG_DESC@
 Version: @PROJECT_VERSION@
 Requires: @PKG_EXTERNAL_DEPS@
 Libs: -L${libdir} @PKG_OMPL_LIBS@
-Cflags: -std=c++14 -I${includedir} -I@Boost_INCLUDE_DIR@
+Cflags: -std=c++14 -I${includedir}
diff --git a/omplConfig.cmake.in b/omplConfig.cmake.in
index 2d88251..e5781c1 100644
--- a/omplConfig.cmake.in
+++ b/omplConfig.cmake.in
@@ -18,8 +18,10 @@ set(OMPL_MAJOR_VERSION @PROJECT_VERSION_MAJOR@)
 set(OMPL_MINOR_VERSION @PROJECT_VERSION_MINOR@)
 set(OMPL_PATCH_VERSION @PROJECT_VERSION_PATCH@)
 
-set_and_check(OMPL_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@")
-set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIR};@Boost_INCLUDE_DIR@;@EIGEN3_INCLUDE_DIR@")
+find_package(Boost COMPONENTS serialization system filesystem)
+
+set_and_check(OMPL_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../include/ompl-${OMPL_MAJOR_VERSION}.${OMPL_MINOR_VERSION}")
+set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIR}")
 foreach(_dir @FLANN_INCLUDE_DIRS@;@ODE_INCLUDE_DIRS@;@SPOT_INCLUDE_DIRS@;@TRIANGLE_INCLUDE_DIR@;@FCL_INCLUDE_DIRS@;@PQP_INCLUDE_DIR@;@ASSIMP_INCLUDE_DIRS@;@OPENGL_INCLUDE_DIR@)
     if(_dir)
         list(APPEND OMPL_INCLUDE_DIRS "${_dir}")
@@ -28,19 +30,13 @@ endforeach()
 list(REMOVE_DUPLICATES OMPL_INCLUDE_DIRS)
 set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIRS}" CACHE STRING "Include path for OMPL and its dependencies")
 
-set_and_check(OMPL_LIBRARY_DIR @PACKAGE_LIB_INSTALL_DIR@)
-set(OMPL_LIBRARY_DIRS "${OMPL_LIBRARY_DIR};@Boost_LIBRARY_DIRS@")
-foreach(_dir @FLANN_LIBRARY_DIRS@;@ODE_LIBRARY_DIRS@;@SPOT_LIBRARY_DIRS@;@FCL_LIBRARY_DIRS@;@ASSIMP_LIBRARY_DIRS@)
-    if(_dir)
-        list(APPEND OMPL_LIBRARY_DIRS "${_dir}")
-    endif()
-endforeach()
-list(REMOVE_DUPLICATES OMPL_LIBRARY_DIRS)
+set_and_check(OMPL_LIBRARY_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../lib")
+set(OMPL_LIBRARY_DIRS "${OMPL_LIBRARY_DIR}")
 set(OMPL_LIBRARY_DIRS "${OMPL_LIBRARY_DIRS}" CACHE STRING "Library path for OMPL and its dependencies")
 
 find_library(OMPL_LIBRARIES NAMES ompl.${OMPL_VERSION} ompl
     PATHS ${OMPL_LIBRARY_DIR} NO_DEFAULT_PATH)
-foreach(_lib @Boost_SERIALIZATION_LIBRARY@;@Boost_FILESYSTEM_LIBRARY@;@Boost_SYSTEM_LIBRARY@;@ODE_LIBRARIES@;@SPOT_LIBRARIES@)
+foreach(_lib Boost::serialization;Boost::filesystem;Boost::system;@ODE_LIBRARIES@;@SPOT_LIBRARIES@)
     if(_lib)
         list(APPEND OMPL_LIBRARIES "${_lib}")
     endif()
diff --git a/src/ompl/CMakeLists.txt b/src/ompl/CMakeLists.txt
index 8e19b30..d9b0cea 100644
--- a/src/ompl/CMakeLists.txt
+++ b/src/ompl/CMakeLists.txt
@@ -59,7 +59,8 @@ target_link_libraries(ompl
     ${Boost_SERIALIZATION_LIBRARY}
     ${Boost_FILESYSTEM_LIBRARY}
     ${Boost_SYSTEM_LIBRARY}
-    ${CMAKE_THREAD_LIBS_INIT})
+    ${CMAKE_THREAD_LIBS_INIT}
+    Eigen3::Eigen)
 
 if (OMPL_EXTENSION_ODE)
     if (NOT CMAKE_VERSION VERSION_LESS 3.13)
-- 
2.41.0

